app/lib/frontend/handlers/documentation.dart

```diff
@@ -43,6 +45,39 @@ Future<shelf.Response> documentationHandler(shelf.Request request) async {
     return redirectResponse('${request.requestedUri}/');
   }
   final String requestMethod = request.method.toUpperCase();
+
+  if (requestContext.experimentalFlags.showSandboxedOutput) {
+    if (requestMethod != 'HEAD' && requestMethod != 'GET') {
+      // TODO: Should probably be "method not supported"!
+      return notFoundHandler(request);
+    }
+
+    final package = docFilePath.package;
+    var version = docFilePath.version!;
+    final path = docFilePath.path!;
+    // Redirect to /latest/ version if the version points to latest version
+    if (version != 'latest') {
```

Comment:
This seems like very similar to the code below on line 86. Maybe merge the two and move up before the experimental conditional check?

Reply:
I tried to keep it separate from the old code, because I want to replace the old logic.